lib/spack/spack/build_systems/autotools.py

```diff
@@ -140,16 +158,46 @@ def patch(self):
 
     def autoreconf(self, spec, prefix):
         """Not needed usually, configure should be already there"""
-        pass
+        # If configure exists nothing needs to be done
+        if os.path.exists(self.configure_abs_path):
+            return
+        # Else try to regenerate it
+        autotools = ['m4', 'autoconf', 'automake', 'libtool']
+        missing = [x for x in autotools if x not in spec]
+        if missing:
+            msg = 'Cannot generate configure: missing dependencies {0}'
+            raise RuntimeError(msg.format(missing))
+        tty.msg('Configure script not found: trying to generate it')
+        tty.warn('*********************************************************')
+        tty.warn('* If the default procedure fails, consider implementing *')
+        tty.warn('*        a custom AUTORECONF phase in the package       *')
+        tty.warn('*********************************************************')
+        with working_dir(self.configure_directory):
+            m = inspect.getmodule(self)
+            m.libtoolize()
+            m.aclocal()
+            m.autoconf()
+            autogen = Executable('./autogen.sh')
+            autogen()
```

Comment:
The last time I tried to generate a configure script, a lot of m4 modules were missing. Autoreconf generated some warnings, but I think it went ok. If possible, we should figure out what is actually necessary to get rid of those warnings. Packages like `libquo`, for example, explicitly define where automake and libtool can be found.